src/firebase/firestore.js

```diff
@@ -1,5 +1,34 @@
-import { getFirestore } from 'firebase/firestore'
-
 import { firebase } from './app'
+import { getFirestore, doc, setDoc, getDoc } from 'firebase/firestore'
 
 export const firestore = getFirestore(firebase)
+
+export const createUserAndFetchDocument = async (user, additionalData) => {
+  if (!user) return null
+
+  const userRef = doc(firestore, 'users', user.uid)
+
+  const snapshot = await getDoc(userRef)
+  if (!snapshot.exists()) {
+    const { email } = user
+    const { firstName, lastName } = additionalData
+
+    try {
+      await setDoc(userRef, {
+        firstName,
+        lastName,
+        email,
+        // ...other data
+      })
+    } catch (error) {
+      console.error('Error creating user document', error)
+      return null
+    }
+  }
+
+  const userDocument = await getDoc(userRef)
+  return { uid: user.uid, ...userDocument.data() }
+}
+
+// const { user } = await createUserWithEmailAndPassword(auth, email, password);
```

Comment:
lets clean up these comments

Reply:
ımm haven't check createUserAndFetchDocument function working or not. but for this part authentication is working fine. I will work on firestore part in different task.